apps/speed.c

```diff
@@ -3950,12 +3950,23 @@ int speed_main(int argc, char **argv)
     }
 
     for (k = 0; k < ALGOR_NUM; k++) {
+        const char *alg_name;
+
         if (!doit[k])
             continue;
+
+        if (k == D_EVP) {
+            if (evp_cipher == NULL)
+                alg_name = evp_md_name;
+            else if ((alg_name = EVP_CIPHER_get0_name(evp_cipher)) == NULL)
+                app_bail_out("failed to get name of cipher '%s'\n", evp_cipher);
```

Comment:
I suspect some compilers with some warnings enabled will complain that `alg_name` can be unused in this function, as `app_bail_out()` doesn't appear to have "no return" annotation

Reply:
Should I change this and set the `alg_name = names[k];` as default? Or would it be okay for these old compilers? And wouldn't these compilers warn about a dereferencing of NULL —if they can find this.

The same problem already exists in line 4576.

Reply:
Yes, that might be better: something like

```
diff --git a/apps/speed.c b/apps/speed.c
index de4f9994db..874899e1e8 100644
--- a/apps/speed.c
+++ b/apps/speed.c
@@ -3952,10 +3952,17 @@ skip_hmac:
     for (k = 0; k < ALGOR_NUM; k++) {
         if (!doit[k])
             continue;
+        const char *alg_name = names[k];
+        if (k == D_EVP) {
+            if (evp_cipher == NULL)
+                alg_name = evp_md_name;
+            else if ((alg_name = EVP_CIPHER_get0_name(evp_cipher)) == NULL)
+                app_bail_out("failed to get name of cipher '%s'\n", evp_cipher);
+        }
         if (mr)
-            printf("+F:%u:%s", k, names[k]);
+            printf("+F:%u:%s", k, alg_name);
         else
-            printf("%-13s", names[k]);
+            printf("%-13s", alg_name);
         for (testnum = 0; testnum < size_num; testnum++) {
             if (results[k][testnum] > 10000 && !mr)
                 printf(" %11.2fk", results[k][testnum] / 1e3);
```

Reply:
> I suspect some compilers with some warnings enabled will complain that `alg_name` can be unused in this function, as `app_bail_out()` doesn't appear to have "no return" annotation

You are really sure this needs to be addressed?  `app_bail_out()` is used so often in the whole file that I can't imagine it being a problem. And in [line 4575](https://github.com/openssl/openssl/blob/2ad07b38ae3da779518149c5f7c1a21b4fada3c2/apps/speed.c#L4575) the code has the same structure without the flaw?

I'm okay with adapting the code, but I question it is needed.

Reply:
I would prefer to see a default value assigned then overwritten as needed, rather than a value assigned in some, but not all, of the `if`-`else` sections

Reply:
Me too.